pkg/plugins/common/postgres/connection.go

```diff
@@ -69,5 +69,13 @@ func ConnectToDbPgx(postgresStoreConfig config.PostgresStoreConfig, customizers
 		return nil, err
 	}
 
-	return pgxpool.NewWithConfig(context.Background(), pgxConfig)
+	p, err := pgxpool.NewWithConfig(ctx, pgxConfig)
+	if err != nil {
+		return nil, err
+	}
+	err = p.Ping(ctx)
```

Comment:
Does this need a retry?

Reply:
Or does the component lifecycle handle restarting

Reply:
It wouldn't. The CP would restart. Yeah, we might need a retry there

Reply:
So you think I should retry around this?